objects/oplayercontroller/Step_0.gml

```diff
@@ -306,6 +383,31 @@ switch state {
 		if !isShortHopping vsp += fallSpeed;
 		else vsp += fastFallSpeed;
 		
+		if (canDoubleJump && !hasSpentDoubleJump)
+		{
+			if (verticalMoveDir == 1 && heldUpFrames <= 1)
+			{
+				vsp = -jumpSpeed;
+				isShortHopping = false;
+				isSuperJumping = false;
+				
+				hsp = walkSpeed * movedir;
+				// Is the player jumping forward?
+				if (movedir = image_xscale)

```

Comment:
Like above, this conditional can be re-written to be one line:

```
isJumpingForward = (movedir == image_scale);
```

But at the very least replace the assignment operator ```=``` with the equivalence one ```==```